### PR TITLE
Improve tests for Annoy index

### PR DESCRIPTION
**Pull request by rschu1ze**
Addresses some suggestions in #50661

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by rschu1ze** [comment]
@devcrafter Maybe you like to check this one?

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 843702ed32b986ab4733c2f6c670bb052501cd1d with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51268/843702ed32b986ab4733c2f6c670bb052501cd1d/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by rschu1ze** [comment]
ClickHouse Integration Tests (asan) [6/6]:
- test_ssl_cert_authentication/test.py::test_https_non_ssl_auth: #49400 

ClickHouse Integration Tests (release) [1/4]:
- test_profile_events_s3/test.py::test_profile_events: #50543

ClickHouse Stateless Tests (msan) [3/6]:
- 00534_functions_bad_arguments12: #50619

ClickHouse Stateless Tests (tsan, S3 Storage) [4/5]:
- 02481_parquet_list_monotonically_increasing_offsets: timed out

File: tests/queries/0_stateless/02354_annoy_index.sql [hunk 1]
```diff
@@ -53,30 +55,34 @@ INSERT INTO tab VALUES (1, [0.0, 0.0, 10.0]), (2, [0.0, 0.0, 10.5]), (3, [0.0, 0
 
 -- See (*) why commented out
 -- SELECT 'WHERE type, L2Distance';
--- SELECT *
+-- WITH [0.0, 0.0, 10.0] AS reference_point
```
Comment (devcrafter) [inline tests/queries/0_stateless/02354_annoy_index.sql:58]: What the point in comment which just duplicates the subsequent query? There is a number of such comments

File: src/Storages/MergeTree/MergeTreeIndexAnnoy.cpp [hunk 2]
```diff
@@ -123,6 +123,11 @@ MergeTreeIndexAggregatorAnnoy<Distance>::MergeTreeIndexAggregatorAnnoy(
 template <typename Distance>
 MergeTreeIndexGranulePtr MergeTreeIndexAggregatorAnnoy<Distance>::getGranuleAndReset()
 {
+    /// Seed random generator used for building the trees in Annoy. All other parameters being equal (granularity trees, search_k), this
+    /// theoretically makes search results deterministic as well. In practice, it is hard to achieve that on SQL level because the index
+    /// structure is also affected by implicit things like the row order or how the column data is split into parts.
+    index->set_seed(42);
```
Comment (devcrafter) [inline src/Storages/MergeTree/MergeTreeIndexAnnoy.cpp:129]: (1) I didn't see that any tests were added which check stable results when using Annoy index? If adding seed still doesn't make it possible - why the do it at all?

(2) If setting seed still make sense then it'd be preferable to add a setting for it

**Comment by rschu1ze** [comment]
No further need for this PR.

**Pull request closed without merging**